apps/src/lib/kits/maker/boards/microBit/MicroBitBoard.js

```diff
@@ -95,34 +89,46 @@ export default class MicroBitBoard extends EventEmitter {
    * @returns {Promise<void>}
    */
   checkExpectedFirmware() {
-    return Promise.resolve()
-      .then(() => this.openSerialPort())
-      .then(serialPort => this.boardClient_.connectBoard(serialPort))
-      .then(() => {
-        // Delay for 0.25 seconds to ensure we have time to receive the firmware version.
-        return delayPromise(250);
-      })
-      .then(() => {
-        if (
-          this.boardClient_.firmwareVersion.includes(MICROBIT_FIRMWARE_VERSION)
-        ) {
-          return Promise.resolve();
-        } else {
-          if (this.boardClient_.firmwareVersion === '') {
-            // Log if we were not able to determine the firmware version in time.
-            firehoseClient.putRecord({
-              study: 'maker-toolkit',
-              study_group: 'microbit',
-              event: 'firmwareVersionTimeout'
-            });
-            console.warn(
-              'Firmware version not detected in time. Try refreshing the page.'
-            );
+    if (!isWebSerialPort(this.port)) {
+      // maker app pathway
+      return Promise.resolve()
+        .then(() => this.openSerialPort())
+        .then(serialPort => this.boardClient_.connectBoard(serialPort))
+        .then(() => {
+          // Delay for 0.25 seconds to ensure we have time to receive the firmware version.
+          return delayPromise(250);
+        })
+        .then(() => {
+          if (
+            this.boardClient_.firmwareVersion.includes(
+              MICROBIT_FIRMWARE_VERSION
+            )
+          ) {
+            return Promise.resolve();
+          } else {
+            if (this.boardClient_.firmwareVersion === '') {
+              // Log if we were not able to determine the firmware version in time.
+              firehoseClient.putRecord({
+                study: 'maker-toolkit',
+                study_group: 'microbit',
+                event: 'firmwareVersionTimeout'
```

Comment:
Are there constants for these somewhere? I'd imagine we probably have other events reported to the `maker-toolkit` study?

Reply:
There was a constant for 'microbit' but I added the other 2 in appropriate files. Thanks!